packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -215,23 +426,23 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
     Rect floatingActionButtonRect;
     if (hasChild(_ScaffoldSlot.floatingActionButton)) {
       final Size fabSize = layoutChild(_ScaffoldSlot.floatingActionButton, looseConstraints);
-      double fabX;
-      assert(textDirection != null);
-      switch (textDirection) {
-        case TextDirection.rtl:
-          fabX = _kFloatingActionButtonMargin + endPadding;
-          break;
-        case TextDirection.ltr:
-          fabX = size.width - fabSize.width - _kFloatingActionButtonMargin - endPadding;
-          break;
-      }
-      double fabY = contentBottom - fabSize.height - _kFloatingActionButtonMargin;
-      if (snackBarSize.height > 0.0)
-        fabY = math.min(fabY, contentBottom - snackBarSize.height - fabSize.height - _kFloatingActionButtonMargin);
-      if (bottomSheetSize.height > 0.0)
-        fabY = math.min(fabY, contentBottom - bottomSheetSize.height - fabSize.height / 2.0);
-      positionChild(_ScaffoldSlot.floatingActionButton, new Offset(fabX, fabY));
-      floatingActionButtonRect = new Offset(fabX, fabY) & fabSize;
+      
+      // To account for the FAB position being changed, we'll lerp between its old and new position.
```

Comment:
let's say something like "we'll animate between" rather than lerp between, because we don't necessarily "lerp" in the sense of calling Offset.lerp.

Reply:
Done